Columns/Metrics/OverallBandwidth.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Bandwidth\Columns\Metrics;
+
+use Piwik\Piwik;
+use Piwik\Plugins\Bandwidth\Metrics;
+
+/**
+ * The max amount bandwidth of a pages.
+ */
+class OverallBandwidth extends Base
+{
+    protected $metric = Metrics::METRICS_PAGE_MAX_BANDWIDTH;
```

Comment:
That doesn't sound correct. Isn't overall actually the same as the sum? But the metric set here is the maximum bandwidth, which should be something else.

I guess this class maybe shouldn't inherit from Base. Actually it's not even a ProcessedMetric, as it doesn't compute a metric. Guess the class is actually only meant for formatting.

I'm not yet sure how to solve that right, but this additional class actually also breaks the overview evolution chart. It's possible to display three bandwidth metrics in there. The overall metric seems now be formatted before while the other two aren't formatted and thus have a lot higher numbers. Not sure how that formatting behaves when having very different numbers.  Wondering if `50 M` would be displayed higher than `5 G` or so. Guess we would need to force using a specific unit (maybe based on the avg number or so).

Reply:
It looks like the bandwidth metrics on the overview evolution chart were broken before this change.  The overall bandwidth figure is formatted but the other two are in bytes. This is from 4.x-dev:
![image](https://user-images.githubusercontent.com/88810029/165870078-c3497029-582a-4811-98b6-00e495ba4542.png)
From what I can see the chart isn't able to compare the different units properly so `5 G` would be seen as smaller than `50 M`.

Seems like this should be addressed as a separate issue from #10716?

I can revert my change to added the `OverallBandwidth` metric which would result in the sparkline value being shown in bytes or maybe it could be left for now as it is correctly formatting the value (albeit using the wrong metric property). The sparkline value is not formatted unless the metric class is a `ProcessedMetric` descendant and `Base` implemented most of what is needed.

Reply:
Being on 4.x-dev in this plugin and in core the numbers are of the same dimension.
![image](https://user-images.githubusercontent.com/1579355/165941127-8ea937f5-3e9d-4e1c-ac9b-2abae34705a4.png)

The formatting is caused by the changes here.

Reply:
Sorry, yes that's correct 4.x-dev shows all chart values in bytes, I'd not switched the plugin branch back to 4.x-dev to test. :facepalm:  It's the change that adds the new metric to `reports/Get.php` which causes the different formatting dimension on the chart.

It looks like there is no good way to have the metric formatted differently for the chart and the sparkline value?

Feel out of scope for this change to start reworking the bandwidth metric formatting. Maybe we don't support sparkline evolution for the overall bandwidth metric for now?

Reply:
@bx80 I've now pushed some changes to this branch. There is now a metric class for all the aggregated metrics in `Get` report. Currently all those metrics a defined as `ProcessedMetric`, which is logically wrong, as the should be `AggregatedMetric`, but we need to add them as processed metrics to the `Get` report in order to have the automatic formatting done by core. This is imho something we should change in core somewhen in the future. Imho there should be a Metric class for every Metric that is used somewhere. And based on the name the formatter should pick and use such metric classes the format the values. But for now it should work this way. Will also push a commit to the core branch to force a specific unit for byte metrics in evolution charts.